src/guide/working-with-pull-requests.md

```diff
@@ -12,7 +12,7 @@ This section also assumes that you have already read [I have a patch, what do I
 
 ## Think once more
 
-All code reviews in OpenJDK are done in public. Once you open your PR for public review the internet can see it and comment on it. Make sure your code is ready for it. Look through your comments, make sure that temporary code is gone, and make sure you have sanitized your method and variable names.
+All code reviews in OpenJDK are done in public. Once you open your PR for public review the internet can see it and comment on it. Make sure your code is ready for it. Look through your comments, make sure that temporary code is gone, and make sure you have sanitized your method and variable names. Also, make sure you understand your code. Why is it working? What are the potential pitfalls? What are the edge-cases? If you haven't already answered all these questions in the mail conversation that preceded this PR, it's likely that you will need to answer them during the review.
```

Comment:
```suggestion
All code reviews in OpenJDK are done in public. Once you open your PR for public review, the internet can see it and comment on it. Make sure your code is ready for it. Look through your comments, make sure that temporary code is gone, and make sure you have sanitized your method and variable names. Also, make sure you understand your code. Why is it working? What are the potential pitfalls? What are the edge-cases? If you haven't already answered all these questions in the mail conversation that preceded this PR, it's likely that you will need to answer them during the review.
```
Should we add a comma to separate the sentences?

Reply:
That doesn't read right to me. Those aren't two different sentences.

Reply:
These are not two independent sentences but these are two dependent clauses: the first one is a conditional clause that is followed by the main clause. Without the comma between ‘review’ and ‘the internet’, one could interpret that ‘the internet’ starts an adjective clause that describes *the review* further, which is not the case. That's exactly what happened to me when I read the entire paragraph thoroughly. With the comma in place, the confusion goes away, the reader has no chance of misinterpreting the sentence: the conditional clause that starts the sentence is explicitly separated from the start of main clause.

In this case, _‘once’_ is a conjunction, similar to _‘if’_, and you've got a comma after the conditional sentence where it goes before the main clause in the last sentence of this very paragraph:

- Once (if) you…, the internet can see…
- If you…, it is likely…

If a condition or an adverbial goes before the subject, there's usually a comma.